modules/tensor_mechanics/src/userobjects/TensorMechanicsPlasticMohrCoulombMulti.C

```diff
@@ -190,45 +200,134 @@ TensorMechanicsPlasticMohrCoulombMulti::activeConstraints(const std::vector<Real
+}
+
+Real
+TensorMechanicsPlasticMohrCoulombMulti::cohesion(const Real internal_param) const
+{
+  return _cohesion.value(internal_param);
+}
+
+Real
+TensorMechanicsPlasticMohrCoulombMulti::dcohesion(const Real internal_param) const
+{
+  return _cohesion.derivative(internal_param);
+}
+
+Real
+TensorMechanicsPlasticMohrCoulombMulti::phi(const Real internal_param) const
+{
+  return _phi.value(internal_param);
+}
+
+Real
+TensorMechanicsPlasticMohrCoulombMulti::dphi(const Real internal_param) const
+{
+  return _phi.derivative(internal_param);
+}
+
+Real
+TensorMechanicsPlasticMohrCoulombMulti::psi(const Real internal_param) const
+{
+  return _psi.value(internal_param);
+}
+
+Real
+TensorMechanicsPlasticMohrCoulombMulti::dpsi(const Real internal_param) const
+{
+  return _psi.derivative(internal_param);
+}
+
+std::string
+TensorMechanicsPlasticMohrCoulombMulti::modelName() const
+{
+  return "MohrCoulombMulti";
+}
+
+
+bool
+TensorMechanicsPlasticMohrCoulombMulti::returnMap(const RankTwoTensor & trial_stress, const Real & intnl_old, const RankFourTensor & E_ijkl,
+                                Real ep_plastic_tolerance, RankTwoTensor & returned_stress, Real & returned_intnl,
+                                std::vector<Real> & dpm, RankTwoTensor & delta_dp, std::vector<Real> & yf,
+                                bool & trial_stress_inadmissible) const
+{
+  if (!_use_custom_returnMap)
+    return TensorMechanicsPlasticModel::returnMap(trial_stress, intnl_old, E_ijkl,
+                                                  ep_plastic_tolerance, returned_stress, returned_intnl,
+                                                  dpm, delta_dp, yf, trial_stress_inadmissible);
+
+  return doReturnMap(trial_stress, intnl_old, E_ijkl, ep_plastic_tolerance, returned_stress, returned_intnl, dpm, delta_dp, yf, trial_stress_inadmissible);
+}
+
+bool
+TensorMechanicsPlasticMohrCoulombMulti::doReturnMap(const RankTwoTensor & trial_stress, const Real & intnl_old, const RankFourTensor & E_ijkl,
+                                Real ep_plastic_tolerance, RankTwoTensor & returned_stress, Real & returned_intnl,
+                                std::vector<Real> & dpm, RankTwoTensor & delta_dp, std::vector<Real> & yf,
+                                bool & trial_stress_inadmissible) const
+{
+  mooseAssert(dpm.size() == 6, "TensorMechanicsPlasticMohrCoulombMulti size of dpm should be 6 but it is " << dpm.size());
 
   std::vector<Real> eigvals;
   RankTwoTensor eigvecs;
-  stress.symmetricEigenvaluesEigenvectors(eigvals, eigvecs);
+  trial_stress.symmetricEigenvaluesEigenvectors(eigvals, eigvecs);
   eigvals[0] += _shift;
   eigvals[2] -= _shift;
 
-  Real sinphi = std::sin(phi(intnl));
-  Real cosphi = std::cos(phi(intnl));
-  Real coh = cohesion(intnl);
-  Real cohcot = coh*cosphi/sinphi;
+  Real sinphi = std::sin(phi(intnl_old));
+  Real cosphi = std::cos(phi(intnl_old));
+  Real coh = cohesion(intnl_old);
+  Real cohcos = coh*cosphi;
```

Comment:
All these "shorthand" variables should be declared const. It makes their intention more clear to the reader (and the compiler!)